libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -849,6 +849,10 @@ bool HTTPClient::connect(void)
 {
     if(_reuse && _canReuse && connected()) {
         DEBUG_HTTPCLIENT("[HTTP-Client] connect: already connected, reusing connection\n");
+
+#if defined(NO_GLOBAL_INSTANCES)
```

Comment:
This is the negation of https://github.com/esp8266/Arduino/blob/29c63506f7146c9fe85dc9286f63b85b421ff573/cores/esp8266/StreamSend.cpp#L354


Should it be this instead ? 
```cpp
#if defined(NO_GLOBAL_INSTANCES) || defined(NO_GLOBAL_STREAMDEV)
```

Reply:
You are right, I thought that because the person explicitly disabled it it would be better for them to define. But moving it to the stack solves the issue.